RoutesGenereren/mainwindow.cpp

```
@@ -0,0 +1,641 @@
+{
+    QDir buildDir(QCoreApplication::applicationDirPath());
+    buildDir.setNameFilters(QStringList{"*.json"});
+    buildDir.setFilter(QDir::Files | QDir::NoDotAndDotDot);
+
+    auto files = buildDir.entryList();
+    m_ui->comboRouteFiles->clear();
+    if (files.isEmpty()) {
+        m_ui->comboRouteFiles->addItem(tr("[geen routes]"));
+    } else {
+        m_ui->comboRouteFiles->addItems(files);
+    }
+}
+
+void MainWindow::refreshSerialPorts()
+{
+    auto ports = QSerialPortInfo::availablePorts();
+    m_ui->comboPorts->clear();
+    for (const auto &info : ports)
+        m_ui->comboPorts->addItem(info.portName());
+    if (ports.isEmpty())
+        m_ui->comboPorts->addItem(tr("[geen poorten]"));
+}
+
+void MainWindow::on_btnRefreshRoutes_clicked()
+{
+    refreshRouteList();
+}
+
+void MainWindow::on_btnRefreshPorts_clicked()
+{
+    refreshSerialPorts();
+}
+
+void MainWindow::on_btnUploadSerial_clicked()
+{
+    // Get JSON file
+    QString fileName = m_ui->comboRouteFiles->currentText();
+    if (fileName.startsWith("[")) {
+        QMessageBox::warning(this, tr("Geen route"),
+                             tr("Genereer eerst routes op tab 1."));
+        return;
+    }
+    QDir buildDir(QCoreApplication::applicationDirPath());
+    QString src = buildDir.filePath(fileName);
+
+    // Get selected port
+    QString portName = m_ui->comboPorts->currentText();
+    if (portName.startsWith("[")) {
+        QMessageBox::warning(this, tr("Geen poort"),
+                             tr("Sluit koffer aan en ververs poorten."));
+        return;
+    }
+
+    // Read JSON
+    QFile f(src);
+    if (!f.open(QIODevice::ReadOnly)) {
+        QMessageBox::critical(this, tr("Fout"),
+                              tr("Kan %1 niet openen").arg(src));
+        return;
+    }
+    QByteArray jsonData = f.readAll();
+    f.close();
+
+    // Open serial port
+    QSerialPort port(portName, this);
+    port.setBaudRate(QSerialPort::Baud9600);

```

this looks like it overwrites settings? is that true?

---

relevant as the device default is 115200, so this is misleading, and likely to cause issues